presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -1098,36 +1099,36 @@ public PlanWithProperties visitUnion(UnionNode node, PreferredProperties parentP
                         ActualProperties.builder()
                                 .global(partitionedOn(desiredParentPartitioning, Optional.of(desiredParentPartitioning)))
                                 .build()
-                                .withReplicatedNulls(parentPartitioningPreference.isNullsAndAnyReplicated()));
+                                .withReplicatedNulls(parentPartitioningProperties.isNullsAndAnyReplicated()));
             }
 
             // first, classify children into partitioned and unpartitioned
-            List<PlanNode> unpartitionedChildren = new ArrayList<>();
-            List<List<Symbol>> unpartitionedOutputLayouts = new ArrayList<>();
+            List<PlanNode> singleNodeChildren = new ArrayList<>();
```

Comment:
nit: `node` is a bit vague, since it can also mean `PlanNode` in this context.

What about `singeNodeDistributionChildren`?  The name is inspired from `SINGLE_DISTRIBUTION` : https://github.com/prestodb/presto/blob/53e45067b912688382dd8797f0b34b714bbea525/presto-main/src/main/java/com/facebook/presto/sql/planner/SystemPartitioningHandle.java#L59


What do you think ? @hellium01 . Am I too picky here? ;)

Reply:
`singleNodeChildren` sounds ok but `Node` can be planNode or WorkerNode so maybe adding Distribution will make it less ambiguous?  Either way is better than current.

Reply:
If it is called `singeNodeDistributionChildren`, than the `distributedChildren` has to be called `multiNodeDistributedChildren`. Also `singleNodeDistributionOutputLayouts` looks a little bit too verbose. I would rather keep `singleNodeChildren`. It doesn't decrease the readability much, as in pair with `singleNode` - `distributed` it is pretty clear what it means. But it decreases wording and keeps code more concise.